graphql_compiler/tests/test_fast_introspection.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2020-present Kensho Technologies, LLC.
+import unittest
+
+from graphql import GraphQLError, GraphQLSchema, get_introspection_query, graphql_sync
+
+from ..fast_introspection import (
+    remove_whitespace_from_query,
+    try_fast_introspection,
+    whitespace_free_introspection_query,
+)
+from .test_helpers import get_schema
+
+
+introspection_query = get_introspection_query()
+
+
+class FastIntrospectionTests(unittest.TestCase):
+    def test_graphql_get_introspection_query(self) -> None:
+        self.assertEqual(
+            whitespace_free_introspection_query, remove_whitespace_from_query(introspection_query),
+        )
+
+    def test_try_fast_introspection_none(self) -> None:
+        schema = get_schema()
+        self.assertEqual(try_fast_introspection(schema, "not the right query"), None)
+
+    def test_try_fast_introspection_equal_graphql_sync(self) -> None:
+        schema = get_schema()
+        execution_result = try_fast_introspection(schema, introspection_query)
+        self.assertIsNotNone(execution_result)
+        if execution_result:
```

Comment:
This is implicitly `if bool(execution_result)` which is going to evaluate to `False` both if `execution_result == None` but also if somehow that object is falsy. That means that this test (and the one below) could pass if the calls return a falsy-but-non-None `ExecutionResult` value — perhaps one with no result rows, depending on the object's `__bool__` method.

I don't think this is intentional — I don't think that such an empty result is meant to be treated as a test success here. I'm assuming this was added because `mypy` didn't recognize that `assertIsNotNone()` will error on a `None` result? If so, let's cast the `Optional` away, or make this be `if execution_result is not None`.

Reply:
Yep, done to appease `mypy`. Added the explicit `is not None`.